app/components/form_components/autocomplete/sort.js

```diff
@@ -1,15 +1,16 @@
 // Returns an array of integers representing the position of each regex match.
 const matchPositions = (string, regexes) => regexes.map(regex => string.search(regex)).filter(i => i >= 0)
 
-const calculateWeight = (name, rawSynonyms, query) => {
+const calculateWeight = (name, synonyms = [], query) => {
   const regexes = query.split(/\s+/).map(word => new RegExp('\\b' + word, 'i'))
-  const synonyms = rawSynonyms || []
 
   const nameMatchPositions = matchPositions(name, regexes)
   const synonymMatchPositions = synonyms.map(synonym => matchPositions(synonym, regexes)).flat()
 
-  // Require all parts of the query to be matched:
-  if (nameMatchPositions.length !== regexes.length && synonymMatchPositions.length !== regexes.length) return 0
+  // Require either all parts of a name to be matched, or all parts of a synonym
+  const allNameMatches = nameMatchPositions.length === regexes.length
+  const allSynonymMatches = synonymMatchPositions.length === regexes.length
```

Comment:
Hmm, I like the naming here but I still don't know if this is correct. Two things:
 - I can see the intention now that we don't want to return a match if one word matches the name and one matches a synonym. But is that the behaviour that we want? I can imagine searching for degrees you might type "Engineering Meng" as a search, which would match the name and the synonym but fail to match because of this behaviour. I don't think this is super common so not a huge deal
 - If a search term has multiple synonyms that the search term matches, this will return 0. Probably should change this line to: 
 ```suggestion
   const allSynonymMatches = synonymMatchPositions.length >= regexes.length
  ```
 ?

Reply:
@AbigailMcP and I spoke yesterday about whether matching across synonyms and name would be good or not. I think we weren't sure either way - but yours is a good example of where it could be helpful.

Reply:
Changed second bullet! I will leave the synonym logic unchanged until we start using in real life and testing.